integration/spark/app/src/test/java/io/openlineage/spark/agent/lifecycle/SparkApplicationExecutionContextTest.java

```diff
@@ -111,14 +120,16 @@ void testRunAndJobAreSet(SparkSession spark) {
       context.start(mock(SparkListenerApplicationStart.class));
       context.end(mock(SparkListenerApplicationEnd.class));
     }
+    verify(eventEmitter, times(2)).emit(lineageEvent.capture());
```

Comment:
I think the test was not working properly without this line added. No lineage events were captured and assertions in the for loop were not called at all.